src/merge_unbounded.rs

```diff
@@ -87,17 +87,27 @@ impl<S> MergeUnbounded<S> {
             }
         }
     }
+
+    /// Returns `true` if there are no streams in the set.
+    pub fn is_empty(&self) -> bool {
+        if self.groups.is_empty() {
+            true
+        } else {
+            self.groups.iter().all(|g| g.streams.is_empty())
+        }
```

Comment:
```suggestion
        self.groups.iter().all(|g| g.streams.is_empty())
```
> An empty iterator returns true.
Should be sufficient